### PR TITLE
Remove some Vision/Node references from the documentation

### PR DESCRIPTION
**Pull request by gblewis1**
Separate PRs will be needed to address specific features such as query, target type, and possibly a final PR to scrub miscellaneous references in the repo such as in scripts or extra Dockerfiles or directories.

### CODE REVIEW

**Comment by gblewis1** [comment]
@nmgaston @yengliong93 please review. doc change only.

File: docs/In-Band Manageability Developer Guide.md [hunk 1]
```diff
@@ -729,7 +648,6 @@ triggering query commands via manifest returns query results through dynamic tel
 |:------------------------|:-------------------------------------|
 | [all]                   | Publish all the  details together    |
 | [fw]                    | Publish firmware details             |
-| [guid]                  | Publish GUID of Vision card          |
 | [hw]                    | Publish hardware details             |
 | [os]                    | Publish operating system details     |
 | [security]              | Publish security details             |
```
Comment (nmgaston) [inline docs/In-Band Manageability Developer Guide.md:653]: should be able to also remove security.  I think that was only for BC.

Comment (gblewis1) [reply docs/In-Band Manageability Developer Guide.md]: @nmgaston removed

**Review by nmgaston — APPROVED** [review]

File: docs/In-Band Manageability Developer Guide.md [hunk 2]
```diff
@@ -15,13 +15,6 @@
...
          5. [Telemetry Agent](#%EF%B8%8Ftelemetry-agent)
          6. [TRTL](#%EF%B8%8Ftrtl)
-   2. [INBM Vision](#inbm-vision)
-         1. [Vision Agent](#%EF%B8%8Fvision-agent)
-            1. [Vision Agent Overall Class Diagram](#vision-agent-overall-class-diagram)
-            2. [Vision Agent Registry Class Diagram](#vision-agent-registry-class-diagram)
-            3. [Vision Agent Xlink Connectivity Class Diagram](#vision-agent-xlink-connectivity-class-diagram)
-         2. [Node Agent](#%EF%B8%8Fnode-agent)
-            1. [Node Agent Class Diagram](#node-agent-class-diagram)
    3. [INBC](#inbc)
```
Comment (yengliong93) [inline docs/In-Band Manageability Developer Guide.md:18]: I think the number index should be 2.

File: docs/In-Band Manageability Developer Guide.md [hunk 3]
```diff
@@ -15,13 +15,6 @@
...
 3. [Run Agents via Source Code](#run-agents-via-source-code)
 6. [Add New Configuration Parameter](#add-new-configuration-parameter)
```
Comment (yengliong93) [inline docs/In-Band Manageability Developer Guide.md:20]: This number index should start at 4.

Comment (gblewis1) [reply docs/In-Band Manageability Developer Guide.md:20]: @yengliong93 fixed

**Review by yengliong93 — APPROVED** [review]

**Pull request merged**